### PR TITLE
ci: fix awslogs test target and add more test scenarios

### PR DESCRIPTION
**Pull request by ningziwen**
*Description of changes:*

* Fixed the existing awslogs ci
* Added more test scenarios
* Added doc for special failure scenario

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: e2e/awslogs_test.go [hunk 1]
```diff
@@ -47,12 +59,193 @@ var testAwslogs = func() {
 				config.WithEndpointResolverWithOptions(customResolver))
 			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
 			cwClient = cloudwatchlogs.NewFromConfig(cfg)
-			cleanupAwslogs(cwClient, testAwslogsGroup, testAwslogsStream)
+			deleteLogGroup(cwClient, testAwslogsGroup)
+			deleteLogGroup(cwClient, nonExistentAwslogsGroup)
```
Comment (ginglis13) [inline e2e/awslogs_test.go:63]: why delete before and after? In case of some failure to delete from a previous run?

File: e2e/awslogs_test.go [hunk 2]
```diff
@@ -47,12 +59,193 @@ var testAwslogs = func() {
...
+				awslogsGroupKey:               testAwslogsGroup,
+				awslogsStreamKey:              nonExistentAwslogsStream,
+				awslogsEndpointKey:            testAwslogsEndpoint,
+				awslogsMultilinePatternKey:    "^" + testAwslogsMultilinePattern,
+				awslogsDatetimeFormatKey:      testAwslogsDatetimeFormat,
+			}
+			creator := cio.BinaryIO(*Binary, args)
+			err := sendTestLogByContainerd(creator, fmt.Sprintf("%s\n%s\n%s\n%s", firstLineLog, secondLineLog, thirdLineLog, forthLineLog))
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			err = validateTestLogsInAwslogs(cwClient, testAwslogsGroup, nonExistentAwslogsStream, fmt.Sprintf("%s\n", firstLineLog))
```
Comment (ginglis13) [inline e2e/awslogs_test.go:118]: should we validate all lines here instead of just the first?

File: e2e/awslogs_test.go (hunk 1 above)
Comment (ningziwen) [reply e2e/awslogs_test.go:63]: Yes. Also want to take best effort to make the test environment clean.

**Review by sam-berning — DISMISSED** [review]

File: e2e/awslogs_test.go [hunk 3]
```diff
@@ -62,28 +258,46 @@ var testAwslogs = func() {
...
+			err := sendTestLogByContainerd(creator, testLog)
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			err = validateTestLogsInAwslogs(cwClient, testAwslogsGroup, testAwslogsStream, []string{testLog})
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
 		})
 	})
 }
 
-func validateTestLogsInAwslogs(client *cloudwatchlogs.Client, logGroupName string, logStreamName string, testLog string) {
+func validateTestLogsInAwslogs(client *cloudwatchlogs.Client, logGroupName string, logStreamName string, testLogs []string) error {
```
Comment (ginglis13) [inline e2e/awslogs_test.go:273]: nit: nonblocking but style preference, call site will look cleaner
```suggestion
func validateTestLogsInAwslogs(client *cloudwatchlogs.Client, logGroupName string, logStreamName string, testLogs ...string) error {
```

**Review by ginglis13 — DISMISSED** [review]

File: e2e/awslogs_test.go (hunk 3 above)
Comment (ningziwen) [reply e2e/awslogs_test.go:273]: I also want to the do the slice length matching so slice may be more intuitive.

File: e2e/awslogs_test.go (hunk 2 above)
Comment (ningziwen) [reply e2e/awslogs_test.go]: Good call. Unfortunately I identified a weird issue during this check. I opened an issue and added TODO to track it. https://github.com/aws/shim-loggers-for-containerd/issues/78

File: e2e/awslogs_test.go (hunk 3 above)
Comment (ginglis13) [reply e2e/awslogs_test.go:276]: ok, FWIW, you can still do length matching:

```go
func main() {
	s := []string{"runfinch", "hello"}
	s1 := "finch"
	s2 := "world"

	strslice(s, s1, s2)
}

func strslice(slc []string, s ...string) {
	fmt.Println(len(slc))
	fmt.Println(len(s))
	fmt.Println(slc)
	fmt.Println(s)
}
```

```
2
2
[runfinch hello]
[finch world]
```

Comment (ningziwen) [reply e2e/awslogs_test.go:276]: Yeah. I just feel the meaning of slice aligns more, which specifies a set arg of similar elements, but in a lower layer than other args in the function. But call site is more like multiple elements in the same layer with other args in the function and we just don't specify the exact number there, which is usually for `..args`.

The set of all the test logs seems in the same layer as other args, like client, group name, stream name. But each test log does not.

Comment (ginglis13) [reply e2e/awslogs_test.go:276]: > Yeah. I just feel the meaning of slice aligns more, which specifies a set arg of similar elements

Totally agree :)

**Review by ginglis13 — APPROVED** [review]

**Review by sam-berning — APPROVED** [review]

**Pull request merged**